version: '3'

services:
    nginx:
        image: nginx:stable-alpine
        container_name: ${PROJECT_NAME}_nginx
        ports:
            - "8081:80"
        volumes:
            - ./app:/var/www/app
            - ./docker/nginx/app.conf:/etc/nginx/conf.d/app.conf
        depends_on:
            - app
            - db
        restart: on-failure
        networks:
            - default

    app:
        build:
            context: .
            dockerfile: ./docker/php/Dockerfile
        container_name: ${PROJECT_NAME}_app
        ports:
            - "9001:9000"
        volumes:
            - ./app:/var/www/app
            - ./docker/php/conf.d/xdebug.ini:/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
            - ./docker/php/conf.d/error_reporting.ini:/usr/local/etc/php/conf.d/error_reporting.ini
        depends_on:
            - db
        restart: always
        networks:
            - default
        user: 1000:1000

    db:
        image: postgres:14-alpine
        container_name: ${PROJECT_NAME}_db
        restart: always
        volumes:
            - postgres_data:/var/lib/postgresql/data
        environment:
            POSTGRES_USER: ${POSTGRES_USER}
            POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
            POSTGRES_DB: ${POSTGRES_DB}
        ports:
            - "35432:5432"
        networks:
            - default

networks:
    default:
        driver: bridge

volumes:
    postgres_data: